### PR TITLE
fix: make sure validation of invalid document exits `1`

### LINKED ISSUE #861: fail-severity warn info
#### Describe the bug
The fail-severity doesn't generate an exit code 1 even for the warn/info severity.  
@asyncapi/cli/0.58.12 linux-x64 node-v18.16.0

#### How to Reproduce
`asyncapi validate schema.yaml --fail-severity=warn`
image
`echo $?`
`0`

#### Expected behavior
The exit code should be `1` but got `0`.

### PR DESCRIPTION
**Pull request by shawakash**
**Description**

- Fixes the issue #861  having validation exit code error by checking the return status
- Added a enum for validation return status

**Related issue(s)**
Resolves #861

### CODE REVIEW

**Comment by shawakash** [comment]
@smoya  Please review it

**Review by MayraYecenia — APPROVED** [review]

**Review by derberg — APPROVED** [review]
lgtm

@Souvikns ?

File: src/commands/validate.ts [hunk 1]
```diff
@@ -30,5 +30,9 @@ export default class Validate extends Command {
     }
 
     await validate(this, specFile, flags);
+    if(await validate(this, specFile, flags) == ValidationStatus.Invalid) {
```
Comment (derberg) [inline src/commands/validate.ts:33]: actually why validate twice?

also `this.exit(0)` is not really necessary

Comment (shawakash) [reply src/commands/validate.ts]: @derberg  changed it

**Comment by Souvikns** [comment]
@shawakash there are some lint issues.

**Comment by peter-rr** [comment]
@Souvikns Should we close this PR as the bug related is already solved by https://github.com/asyncapi/cli/pull/1068 ?

**Comment by Souvikns** [comment]
We can close it, but this PR adds some refactoring, that we can use.

**Comment by peter-rr** [comment]
> We can close it, but this PR adds some refactoring, that we can use.

@Souvikns 
Yeah, we can keep it as a refactor then. Should we change the title from `fix:` to `refactor:` in that case?

@shawakash 
Do you need some help to move forward with this PR?

**Pull request closed without merging**